watchtower/wtpolicy/policy.go

```diff
@@ -114,20 +116,44 @@ type Policy struct {
 }
 
 // String returns a human-readable description of the current policy.
-func (p Policy) String() string {
+func (p *Policy) String() string {
 	return fmt.Sprintf("(blob-type=%b max-updates=%d reward-rate=%d "+
 		"sweep-fee-rate=%d)", p.BlobType, p.MaxUpdates, p.RewardRate,
 		p.SweepFeeRate)
 }
 
+// FeatureBits returns the watchtower feature bits required for the given
+// policy.
+func (p *Policy) FeatureBits() []lnwire.FeatureBit {
+	features := []lnwire.FeatureBit{
+		wtwire.AltruistSessionsRequired,
+	}
+
+	t := p.TxPolicy.BlobType
+	switch {
+	case t.IsTaprootChannel():
+		features = append(features, wtwire.TaprootCommitRequired)
+	case t.IsAnchorChannel():
+		features = append(features, wtwire.AnchorCommitRequired)
+	}
+
+	return features
+}
+
 // IsAnchorChannel returns true if the session policy requires anchor channels.
-func (p Policy) IsAnchorChannel() bool {
+func (p *Policy) IsAnchorChannel() bool {
```

Comment:
wondering whether we should restrict the method `IsAnchorChannel`, and all its occurrence, to only return true if this channel is anchor type, and not taproot type - should help to avoid misusing as we now have to always keep in mind to check `IsTaproot` first before check `IsAnchor` in a switch statement, maybe we should do this on `ChannelType` - future PRs tho.

Reply:
> so for this instance (ie, in towers alone) it does actually just mean anchors. So this will return false for a taproot blob type.

Yeah I do notice that the feature bit is `101` and `11`, tho not sure about how to properly define a feature bit, as it seems that technically if anchor is a feature, then taproot chan should be `111` since it has this feature - unknown implications here, but I prefer the end result here as we can clearly distinguish the two channels. Meanwhile this shouldn't be an issue for feature negotiation since both are `lnd`? I think if both sides understand and use the same defined feature bits we are fine, eg, they won't send us a `111`.

Reply:
yeah i guess it's slightly different in this tower protocol cause the tower only cares about commit types as a whole since this completely changes the justice kit etc. 

> Meanwhile this shouldn't be an issue for feature negotiation since both are lnd?

yeah - currently the code for this _is_ the spec. So this is fine for 2 lnd nodes, yeah 👍